articles/introduction-to- virtual- environments-and-dependency-managers/index.md

```diff
@@ -0,0 +1,419 @@
+﻿### Introduction to Python Virtual Environments and Dependency managers
+
+Virtual environments. What are they and what are they used for? From the Python documentation, a virtual environment is an isolated Python environment where a Python interpreter, libraries, modules and scripts (collectively called dependencies) are installed in a different directory from those installed in the system’s default Python path and other virtual environments.
+
+Dependency managers are tools that enable easy management of a project's dependencies.
+
+### Prerequisites
+
+This article assumes:
+
+* The reader's knowledge of Python programming language.
+* Has Python installed into his/her computer.
+
+    >Just to make sure run this command on your terminal.  `python --version`
+
+### Overview
+
+This article will look into the most used virtual environments and dependency managers in python, how to use them and their pros and cons.
+
+1. [Pipenv](#1.-Pipenv)
+2. [Venv](#2.-Venv)
+3. [Virtualenv](#3.-Virtualenv)
+4. [Poetry](#4.-Poetry)
+5. [Conda](#5.-Conda)
```

Comment:
Rearrange these to follow the order: `2 > 3 > 1 > 5 > 4` ie. popularity.

Reply:
I have corrected the changes requested.